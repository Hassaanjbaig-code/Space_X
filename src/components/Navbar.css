.navbar {
  justify-content: space-between;
  padding: 12px;
  
}

.navbar img {
  width: 40px;
  height: 40px;
  margin-right: 12px;
}

.list-nav {
  gap: 8px;
  font-size: 0.8rem;
  color: var(--color-text-blue);
}

.seprator {
  height: 15px;
  border: 1px solid #121212;
}
