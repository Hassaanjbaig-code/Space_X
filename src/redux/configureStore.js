import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketReducer';
import missionReducer from './missions/missionReducer';

export default configureStore({
  reducer: {
    rocketReducer,
    mission: missionReducer.reducer,
  },
});
