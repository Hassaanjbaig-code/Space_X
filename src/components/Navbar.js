import React from 'react';
import logo from '../assets/images/planet.png';
import './Navbar.css';

function Navbar() {
  return (
  <nav className='flex-center navbar'>
    <div className="row logo">
      <img src={logo} alt="Logo Space X" />
      <h1>Space Travelers'Hub</h1>
    </div>
    <ul className='flex-center no-style list-nav'>
      <li>Rockets</li>
      <li>Missions</li>
      <hr className='seprator'/>
      <li>My Profile</li>
    </ul>
  </nav>
  )
}

export default Navbar