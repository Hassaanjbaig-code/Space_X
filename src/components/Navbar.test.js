
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux';
import React from 'react';
import Navbar from './Navbar';
import configureStore from '../redux/configureStore';
import { BrowserRouter as Router } from 'react-router-dom';

describe('test render', () => {
  test('List Rocket should match snapshoot', () => { 
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={configureStore}>
          <Router>
            <Navbar />
          </Router>
        </Provider>
      </React.StrictMode>,
    ).toJSON()
    expect(tree).toMatchSnapshot();
  })
})