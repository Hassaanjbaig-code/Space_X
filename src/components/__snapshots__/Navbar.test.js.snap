// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<nav
  className="flex-center navbar"
>
  <div
    className="row logo"
  >
    <img
      alt="Logo Space X"
      src="planet.png"
    />
    <h1>
      Space Travelers Hub
    </h1>
  </div>
  <ul
    className="flex-center no-style list-nav"
  >
    <li>
      <a
        aria-current="page"
        className="navLink active"
        href="/"
        onClick={[Function]}
        path="/"
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="navLink"
        href="/Missions"
        onClick={[Function]}
        path="/Missions"
      >
        Missions
      </a>
    </li>
    <hr
      className="seprator"
    />
    <li>
      <a
        className="navLink"
        href="/My Profile"
        onClick={[Function]}
        path="/My Profile"
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
